src/main/java/org/riders/sharing/service/impl/ScooterServiceImpl.java

```
@@ -0,0 +1,40 @@
+package org.riders.sharing.service.impl;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.riders.sharing.dto.PageRequestDto;
+import org.riders.sharing.dto.PageResponseDto;
+import org.riders.sharing.dto.ScooterDto;
+import org.riders.sharing.exception.NoElementException;
+import org.riders.sharing.repository.ScooterRepository;
+import org.riders.sharing.service.ScooterService;
+
+import java.util.ArrayList;
+
+public class ScooterServiceImpl implements ScooterService {
+    private static final Logger logger = LogManager.getLogger(ScooterServiceImpl.class);
+    private final ScooterRepository scooterRepository;
+
+    public ScooterServiceImpl(ScooterRepository scooterRepository) {
+        this.scooterRepository = scooterRepository;
+    }
+
+    @Override
+    public PageResponseDto<ScooterDto> getAvailableScooters(PageRequestDto requestDto) {
+        final var scooterDtoList = new ArrayList<ScooterDto>();
+        final var page = requestDto.page();
+        final var pageSize = requestDto.size();
+        final var offset = (page - 1) * pageSize;
+        final var totalElements = scooterRepository.getAvailableScootersAmount();
+        final var totalPages = (int) Math.ceil((double) totalElements / pageSize);
+        final var availableScooterList = scooterRepository.findAvailableScootersForResponse(pageSize, offset);
+
+        if (availableScooterList.isEmpty()) {
+            logger.error("Couldn't find available scooters");
+            throw new NoElementException("Couldn't find available scooters");
```

do we really want to throw an exception in the case if the available scooters are not found?
imho it's okey from the business logic perspective. For end user it will look weird to see an exception, I believe it's better to return an empty response and create a notification message says "Currently no bikes available, please try again later" or something similar.

In general, returning an error to the user is the last thing you should do. If there is a way to process the request somehow, it is better to do so. And return an error only if the business logic requires it or if there is no other way to process it at runtime.